crates/language_models/src/provider/open_ai.rs

```diff
@@ -557,6 +557,15 @@ impl OpenAiEventMapper {
             }
         }
 
+        if let Some(usage) = event.usage {
```

Comment:
We should also check for usage if we return early (above, when event.choices is empty). Seen this recently when using v0 where the usage stats where streamed in last (and event.choices was empty)

Reply:
Fixed this to raise the usage event as soon as we get one before processing the event stream.